### PR TITLE
Fix integration slow tests

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR fixes the following slow tests which are failing because of the change of beahvior in the `Embeddings` layer in PyTorh 1.8. This is done by adding an attention mask to ignore the padding token and checking a slice that does not contain the padding hidden states.
```
tests/test_modeling_albert.py::AlbertModelIntegrationTest::test_inference_no_head_absolute_embedding
tests/test_modeling_bert.py::BertModelIntegrationTest::test_inference_no_head_absolute_embedding
tests/test_modeling_bert.py::BertModelIntegrationTest::test_inference_no_head_relative_embedding_key
tests/test_modeling_bert.py::BertModelIntegrationTest::test_inference_no_head_relative_embedding_key_query
tests/test_modeling_convbert.py::ConvBertModelIntegrationTest::test_inference_masked_lm
tests/test_modeling_deberta.py::DebertaModelIntegrationTest::test_inference_no_head
tests/test_modeling_deberta_v2.py::DebertaV2ModelIntegrationTest::test_inference_no_head
tests/test_modeling_distilbert.py::DistilBertModelIntergrationTest::test_inference_no_head_absolute_embedding
tests/test_modeling_electra.py::ElectraModelIntegrationTest::test_inference_no_head_absolute_embedding
tests/test_modeling_squeezebert.py::SqueezeBertModelIntegrationTest::test_inference_classification_head
```

It also fixes
```
tests/test_modeling_mbart.py::MBartEnroIntegrationTest::test_enro_generate_batch
```
that was failing since the change in `prepare_seq2seq_batch`. For some reason a word is different but it was consistent in PyTorch 1.7/PyTorch 1.8 so I changed the desired target. @patil-suraj if you want to take a closer look, I'll leave it to you.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
This looks good to me, thanks for taking care of it!
